ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -160,8 +164,11 @@ initialChainSelection immutableDB volatileDB lgrDB tracer cfg varInvalid
     constructChains i succsOf = flip evalStateT Map.empty $
         mapM constructChain suffixesAfterI
       where
+        -- TODO ok to avoid selecting blocks more than k past the imm tip during
+        -- initial chain selection? (also, we currently don't use cdbLoELimit
+        -- here)
         suffixesAfterI :: [NonEmpty (HeaderHash blk)]
-        suffixesAfterI = Paths.maximalCandidates succsOf (AF.anchorToPoint i)
+        suffixesAfterI = Paths.maximalCandidates succsOf k (AF.anchorToPoint i)
```

Comment:
> Why don't we use `cdbLoELimit` here?

It is not in scope here directly (though it could be passed in), and it didn't seem that relevant for the prototype, so I just went with `k` as it seemed to capture what initial chain selection is about intuitively (re-establish a volatile selection of length `k` after startup). I also agree with your intuition that using something $<k$ seems fine; it seems similar to a VolatileDB corruption, which is something we already allude to in several places and intend to handle just fine.